scripts/src/sanitycheckpr/sanitycheckpr.py

```diff
@@ -17,23 +23,71 @@ def ensure_only_chart_is_modified(api_url):
     files_api_url = f'{api_url}/files'
     headers = {'Accept': 'application/vnd.github.v3+json'}
     r = requests.get(files_api_url, headers=headers)
-    pattern = re.compile(r"charts/(\w+)/([\w-]+)/([\w-]+)/([\w\.]+)/.*")
+    pattern = re.compile(r"charts/(\w+)/([\w-]+)/([\w-]+)/(.+?(?=\/))/.*")
```

Comment:
the current and proposed code changes still make possible that a PR contains changes to multiple charts. Namely, the regex only looks if filepaths are starting with `charts/` prefix and have the defined depth.

IMHO, forbidding updates can be done a way simpler, even without checking index.yaml:

* Get list of files submitted in PR (that is already done in line 25)
* All file statuses should be set to `added` - otherwise, we can exit with error - if PR contains some files that are modified, it means that PR modifies some existing chat, and that chart is then already published, because it is a part of code repo.
* All files in the PR should have share the same common prefix

There is no need to download any `index.yaml` and the whole workflow logic becomes a way simpler.

Reply:
cc @mmulholla

Reply:
Done. 5d5ac7e4ef53852d1935ce45c92b993980b7884d

Reply:
It is possible to update a chart by adding a new file,  which this would now miss. Not sure if you can add a file which could affect the contents of the Index.yaml file but we would continue to create the release. The previous approach required more code to cover this issue.

Reply:
maybe we just add a check that Chart.yaml is included in the PR?

Reply:
Added checking the existence of chart and/or report as part of the PR:
0f8447bdd008cb36e49214fbf310db4e69feea04

Reply:
I don't think  the logic is quite right yet.

report found, need to ensure no other files are added unless chart.yaml is present.  
report found, tarball found, need to ensure no other files are added.
no report, must ensure either chart.yaml is included, or ,tgz is the only file.

Reply:
These combination checks are present as part of `Verify PR - generate report`  step.
In the sanity check, we need to ensure only addition is happening. And you pointed what if the user adds an extra file to the existing chart. To last commit ensure "Chart.yaml" is in the add "added" list (Same with report and chart).  This was we can ensure only addition is part of the PR. And if it's all addition there is an addition to one of the above three files.

Reply:
I thought the idea here was to prevent updating an existing chart by detection in sanity check.
1. I can publish a chart without a report and then add a report later, this should be caught but is not in the current logic. 
2. I can publish a chart using just a report, then add chart source or a tarball, this should be caught but is not in the current logic.   
3. Not sure what it means if I publish a chart using source and then try adding a tarball, but should probably also be caught in this logic.

Beginning to think this concept of making sure all contents of a PR are additions is flawed, ad the old logic was better.

Reply:
@mmulholla I agree that this logic is flawed! I am with going back to the old approach in PR.
@pedjak Any comments?

Reply:
good catch @mmulholla 

In order to fix the mentioned issues, we should add one more check:

* the chart directory must not exist in the target branch

Reply:
@mmulholla @pedjak @dperaza4dustbit This discussion is going back and forth multiple times. I need a confirmation whether I should revert my changes as suggested by @mmulholla or try the last suggestion given by @pedjak .

Reply:
Personally I prefer checking Index.yaml content because we are checking exactly what we are trying to avoid, rather than checking some upstream status only to find some corner case that it does not cover.  Checking the chart version directory does not already exist certainly makes for simpler code but I would worry that we could merge a chart and then fail in the release process for some reason. In this case one solution we would now prevent is  to trigger the release process again using a new PR with a irrelevant change.  I am assuming that chart content which passes build and verify could never cause a release proces failure.

Reply:
@predrag please respond to my comment. Will leave the final decision to you.

Reply:
> please respond to my comment. Will leave the final decision to you.

I am fine if you wanna take this route.

Reply:
Why don't we define a set of test cases we want to pass to call this PR done. Then we check against that and move pending things to a different story.